src/Field/Definition/AbstractField.php

```diff
@@ -104,6 +106,16 @@ public function useAsAdminDisplayName (
 		return $this;
 	}
 
+	/**
+	 * Uses this field as admin display preview value
+	 *
+	 * @return $this
+	 */
+	public function useAsAdminDisplayImage () : static
+	{
+		throw new InvalidFieldConfigurationException("This field type cannot be used as admin display image. Only 'AssetField's can be used.");
```

Comment:
I'd love to implement throwing an exception for `useAsAdminDisplayName` for incompatible types as well, though this might be considered a breaking change.

It is a bug, though, and not what I'm as a developer expecting. I'd much rather see an exception as soon as I try to sync/validate my components and get notified that I did a boo-boo, instead of thinking it's working and most likely not even noticing it.

Reply:
Add the exception there as well.

Reply:
Added!